### ISSUE TITLE
subsetTax (and likely others) may cause integer overflows

### ISSUE THREAD

**Issue by fpusan** [jtamames/SqueezeMeta#138]
This happens when working with very big datasets.
`1: In FUN(X[[i]], ...) : integer overflow - use sum(as.numeric(.))`
Likely fix would be to explicitly cast numeric columns to the numeric type when creating the SQM objects.

**Comment by bresyd** [comment]
Hi Fernando, 

I just stumbled across an issue after using the subsetORFs function, maybe it is related to the error that you reported here.

I am trying to create a combined table containing the ORF, contig and bin information for some ORFs of interest, but there is an issue with the Bin IDs.

Here is what I do:
```

# make a vector containing the orf names
test_orfs <- c("megahit_363190_1-96", "megahit_413329_895-1317", "megahit_458224_2-187", "megahit_605321_2-169", "megahit_733519_1958-2851", "megahit_966690_1138-1290", "megahit_1978256_3-206", "megahit_2338645_3-491", "megahit_2342939_23546-23749", "megahit_2503433_63512-64405", "megahit_2512696_20560-20796")
# make subset with test_orfs
test_subset = subsetORFs(so245, orfs = test_orfs)

test_bin_table = test_subset$bins$table
row.names(test_bin_table)
```
[1] "concoct.28.fa_sub.contigs"     "maxbin.0783.fasta_sub.contigs" "maxbin.0345.fasta_sub.contigs" "metabat2.1630.fa_sub.contigs" 
[5] "metabat2.430.fa.contigs"

`test_subset$contigs$bins`
               DASTool                        
megahit_363190  "No_bin"                       
megahit_413329  "No_bin"                       
megahit_458224  "No_bin"                       
megahit_605321  "No_bin"                       
megahit_733519  "concoct.28.fa_sub.contigs"    
megahit_966690  "No_bin"                       
megahit_1978256 "maxbin.0783.fasta_sub.contigs"
megahit_2338645 "maxbin.0345.fasta_sub.contigs"
megahit_2342939 "metabat2.1630.fa_sub.contigs" 
megahit_2503433 "maxbin.0345.fasta_sub.contigs"
megahit_2512696 "metabat2.430.fa.contigs"  

`test_subset$contigs$table$`Bin ID``
[1] ""                              ""                              ""                              ""                             
 [5] "concoct.28.fa_sub.contigs"     ""                              "maxbin.0783.fasta_sub.contigs" "maxbin.0345.fasta_sub.contigs"
 [9] "metabat2.1630.fa_sub.contigs"  "maxbin.0345.fasta_sub.contigs" "metabat2.430" 

Out of some reason, the name of one bin is different in `test_subset$contigs$table$`Bin ID`` compared to `test_subset$contigs$bins` and `test_subset$bins$table`, metabat2.430 instead of the correct metabat2.430.fa.contigs.

This becomes a problem when I try to create a combined table:

```
# now merge the contigs table with the bin table
test_subset_contig_bin_table <- merge.data.frame(test_subset$contigs$table, test_subset$bins$table, all.x = TRUE, by.x="Bin ID", by.y = 'row.names')
subset_table <- subset(test_subset_contig_bin_table, select = c('Bin ID', 'Tax.y', 'Num contigs', 'Completeness'))
subset_table
```
                       
 ```
 Bin ID                                                                                  Tax.y Num contigs Completeness
1                                                                                                                  <NA>          NA           NA
2                                                                                                                  <NA>          NA           NA
3                                                                                                                  <NA>          NA           NA
4                                                                                                                  <NA>          NA           NA
5                                                                                                                  <NA>          NA           NA
6      concoct.28.fa_sub.contigs                                                                             k_Bacteria        1502        35.53
7  maxbin.0345.fasta_sub.contigs         k_Bacteria;p_Proteobacteria;n_delta/epsilon subdivisions;c_Deltaproteobacteria         915        57.73
8  maxbin.0345.fasta_sub.contigs         k_Bacteria;p_Proteobacteria;n_delta/epsilon subdivisions;c_Deltaproteobacteria         915        57.73
9  maxbin.0783.fasta_sub.contigs       k_Bacteria;n_PVC group;p_Verrucomicrobia;c_Verrucomicrobiae;o_Verrucomicrobiales        1562        54.34
10  metabat2.1630.fa_sub.contigs k_Bacteria;p_Proteobacteria;c_Alphaproteobacteria;o_Rhodobacterales;f_Rhodobacteraceae         154        35.85
11                  metabat2.430                                                                                   <NA>          NA           NA

```
Obviously, because the bin IDs are different in the two tables I want to merge, there is no bin information in the merged table for bin metabat2.430.fa.contigs.

Do you have an idea about what goes wrong here and how I can potentially fix it?

Thanks a lot

**Comment by fpusan** [comment]
Hi!
It is indeed weird that the nomenclature is not consistent.
Can you send me the `/path/to/yout/project/results/19.*.bintable` file?
Best,
Fernando

**Comment by fpusan** [comment]
Moved the discussion on bin nomenclature to its own issue

**Comment by fpusan** [comment]
Fixed in 440a524
